src/Stratis.Bitcoin.Tests/Consensus/ChainedHeaderTreeTest.cs

```diff
@@ -150,5 +155,96 @@ public void ConnectHeaders_NewAndExistingHeaders_ShouldCreateNewHeaders()
             Assert.Equal(listOfNewHeaders.Last(), connectedHeadersNew.DownloadTo.Header);
             Assert.Equal(listOfNewHeaders.First(), connectedHeadersNew.DownloadFrom.Header);
         }
+
+        // Supply headers that we already have, make sure no new ChainedHeaders were created.
```

Comment:
For now for simplicity please set the issue test case number on the summery (we can delete later)

Reply:
Use XML comments for commenting on methods.

Reply:
Also feel free to write those descriptions as the descriptions are just quickly hacked and don't have production quality.

Reply:
Updated to XML comment style, descriptions corrected.